lib/config/migrations.go

```diff
@@ -112,7 +114,25 @@ func migrateToConfigV38(cfg *Configuration) {
 		cfg.GUI.WebauthnRpId = defaultGuiCfg.WebauthnRpId
 	}
 	if cfg.GUI.WebauthnOrigin == "" {
-		cfg.GUI.WebauthnOrigin = defaultGuiCfg.WebauthnOrigin
+		_, port, err := net.SplitHostPort(cfg.GUI.RawAddress)
+		if err == nil {
+			scheme := "https"
+			if !cfg.GUI.RawUseTLS {
+				scheme = "http"
+			}
+
+			port = ":" + port
+			if scheme == "https" && port == ":443" {
+				port = ""
+			} else if scheme == "http" && port == ":80" {
+				port = ""
+			}
+
+			cfg.GUI.WebauthnOrigin = scheme + "://" + cfg.GUI.WebauthnRpId + port
+
```

Comment:
You really like blank lines, don't you?  Looking at the surrounding code, this really sticks out.  Please try to use more sparingly, only where the code definitely commences doing something else.  Of course, this is just my opinion.

Reply:
Done in f915d1a095cd4673dbfd629cf49a05eda15ede9a.